### PR TITLE
Feature/add notes to api

### PR DESCRIPTION
**Pull request by jmassie**
added notes to v2 and migration files in api - Scottish-Natural-Heritage/Licensing#1357

### CODE REVIEW

File: src/controllers/v2/note.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+const NoteController = {
+  /**
+   * The create function writes the incoming Note to the appropriate database tables.
+   *
+   * @param {any } appId The application that the Note will be based on.
+   * @param {any | undefined} incomingNote The Note details.
+   * @returns {any} Returns newNote, the newly created Note.
+   */
+  // prettier-ignore
```
Comment (paulcoxsnh) [inline src/controllers/v2/note.js:13]: You don't need to disable this linting rule, or the other 3 `// prettier-ignore` exceptions you've added to this file.

File: src/controllers/v2/note.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    let newNote;
+    // Start a transaction.
+    await database.sequelize.transaction(async (t) => {
+      incomingNote.ApplicationId = appId;
+
+      // Add the Note to the DB.
+      newNote = await Note.create(incomingNote, {transaction: t});
+    });
+
+    // If all went well and we have a new application return it.
```
Comment (paulcoxsnh) [inline src/controllers/v2/note.js:24]: This comment doesn't make sense, double check copy pasted comments, there are a couple in this file that are wrong.

File: src/controllers/v2/note.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    // If no new application was added to the DB return undefined.
+    return undefined;
+  },
+
+  /**
+   * Retrieve the notes from the database.
+   *
+   * @param {number} id An existing sett's ID.
```
Comment (paulcoxsnh) [inline src/controllers/v2/note.js:36]: Setts ID?

File: util/db/migrations/20220623082412-sfo-create-notes.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      deletedAt: {
+        type: Sequelize.DATE
+      }
+    });
+  },
+  // prettier-ignore
```
Comment (paulcoxsnh) [inline util/db/migrations/20220623082412-sfo-create-notes.js:37]: Ditto.

File: src/controllers/v2/note.js (hunk 1 above)
Comment (jmassie) [reply src/controllers/v2/note.js:13]: Done

Comment (jmassie) [reply src/controllers/v2/note.js:13]: Done

File: src/controllers/v2/note.js (hunk 2 above)
Comment (jmassie) [reply src/controllers/v2/note.js:24]: Done

File: src/controllers/v2/note.js (hunk 3 above)
Comment (jmassie) [reply src/controllers/v2/note.js:36]: Done

File: util/db/migrations/20220623082412-sfo-create-notes.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+module.exports = {
+  // prettier-ignore
```
[the remark this answers is not in the document]
Comment (jmassie) [reply util/db/migrations/20220623082412-sfo-create-notes.js:2]: Done

File: util/db/migrations/20220623082412-sfo-create-notes.js (hunk 4 above)
Comment (jmassie) [reply util/db/migrations/20220623082412-sfo-create-notes.js:37]: Done

File: src/controllers/v2/note.js (hunk 1 above)
Comment (paulcoxsnh) [reply src/controllers/v2/note.js]: Resolved in https://github.com/Scottish-Natural-Heritage/standard-forestry-operations-api/pull/77/commits/e33b7df9f6afa48d4e7d2ef0bd9782a3a13aa654

File: src/controllers/v2/note.js (hunk 3 above)
Comment (paulcoxsnh) [reply src/controllers/v2/note.js]: Resolved in https://github.com/Scottish-Natural-Heritage/standard-forestry-operations-api/pull/77/commits/e33b7df9f6afa48d4e7d2ef0bd9782a3a13aa654

File: util/db/migrations/20220623082412-sfo-create-notes.js (hunk 4 above)
Comment (paulcoxsnh) [reply util/db/migrations/20220623082412-sfo-create-notes.js]: Resolved in https://github.com/Scottish-Natural-Heritage/standard-forestry-operations-api/pull/77/commits/e33b7df9f6afa48d4e7d2ef0bd9782a3a13aa654

File: util/db/migrations/20220623082412-sfo-create-notes.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (paulcoxsnh) [reply util/db/migrations/20220623082412-sfo-create-notes.js]: Resolved in https://github.com/Scottish-Natural-Heritage/standard-forestry-operations-api/pull/77/commits/e33b7df9f6afa48d4e7d2ef0bd9782a3a13aa654

File: src/controllers/v2/note.js (hunk 2 above)
Comment (paulcoxsnh) [reply src/controllers/v2/note.js]: Resolved in https://github.com/Scottish-Natural-Heritage/standard-forestry-operations-api/pull/77/commits/e33b7df9f6afa48d4e7d2ef0bd9782a3a13aa654

**Review by paulcoxsnh — APPROVED** [review]
All good - tested in Docker and it seems to do what it's supposed to :)

**Pull request merged**